lib/pka_dev.c

```diff
@@ -1764,53 +1552,10 @@ static int pka_dev_init_shim(pka_dev_shim_t *shim)
         return -EPERM;
     }
 
-    // First of all, trigger a hardware zeroize to initialize internal
-    // RAM memories
-    ret = pka_dev_ram_zeroize(&shim->resources.ext_csr);
-    if (ret)
-    {
-        PKA_ERROR(PKA_DEV, "failed to zeroize PKA\n");
-        return ret;
-    }
-
-    // Configure AIC registers
-    ret = pka_dev_config_aic_interrupts(&shim->resources.aic_csr);
-    if (ret)
-    {
-        PKA_ERROR(PKA_DEV, "failed to configure AIC\n");
-        return ret;
-    }
-
-    shim_fw_id = pka_firmware_get_id();
-
-    // Load Farm image into PKA_BUFFER_RAM for non-High Assurance mode
-    // or into PKA_SECURE_RAM for High Assurance mode.
-    farm_img_size = pka_firmware_array[shim_fw_id].farm_img_size;
-    PKA_DEBUG(PKA_DEV, "loading farm image (%d words)\n", farm_img_size);
-
-    farm_img_ptr = pka_firmware_array[shim_fw_id].farm_img;
-    // The IP provider suggests using the zeroize function to initialize
-    // the Buffer RAM. But a bug has been detected when writing ECC bits.
-    // Thus a workaround is used, and has already been shown to work; it
-    // consists of padding the farm image. Then all RAM locations will be
-    // written with correct ECC before the IP reads the image out.
-    ret = pka_dev_load_image(&shim->resources.buffer_ram, farm_img_ptr,
-                                farm_img_size);
-    if (ret)
-    {
-        PKA_ERROR(PKA_DEV, "failed to load farm image\n");
-        return ret;
-    }
-
-    // Configure EIP-154 Master controller Sequencer
-    ret = pka_dev_config_master_seq_controller(shim,
-                                        &shim->resources.master_seq_ctrl);
-    if (ret)
-    {
-        PKA_ERROR(PKA_DEV, "failed to configure Master controller "
-                                    "Sequencer\n");
-        return ret;
-    }
+    // The folloiwng initialization tasks has been completed in UEFI:
+    // - Configure AIC registers
+    // - Load PKA firmware
+    // - Configure master controller sequencer

```

Comment:
I believe this is irrelevant. I suggest we omit this comment

Reply:
Those are removed in the latest update.